### ISSUE TITLE
kill Website Language ? - decision needed by Wed Mar 11

### ISSUE THREAD

**Issue by ezoehunt** [MarketplaceUX/mow#4]
### Decision needed by Wed Mar 11 !

The original thinking is that it is useful for consumers to know that a website is visible in a particular language. E.g., the website detail page would say "Available in English, Spanish, and French." 

I have been re-thinking this and now think that it's a lot of overhead that produces little value. Let me explain.
- Accessing an app is an activity with friction. That is, getting an app means you have to download, then install, then launch. 
- For that reason, it's useful for consumers to have a lot of information to inform their "purchase" decision. That is, the more information I have on the app detail page, the better informed I will be to make a choice to initiate the install app process (or not). 
- Language is one of the important ways to determine app relevance to me. It is a huge bummer to install an app and open it to discover that it's not in your language.
- Accessing a website, on the other hand, is a single-click activity. When you open the website to find it's in another language, you close the tab (or translate if you're using Chrome :)

There's so little friction attached to clicking a website link that I'm inclined to remove language from the website detail page. 

It will be complicated for us to capture the "languages" that a website might be localized into. E.g., when a submitter submits a website, he will only be able to tell us the language he sees on the screen. He won't be able to tell us all the languages it's available in. 
- Unless we ask him to change his browser language and test for every language in the world (silly idea). 
- Or unless we want to build some sort of tool to do this.
- Or some other solution I haven't thought of, but sounds like a lot of work

In short, I see "website language" as a lot of work for us with little consumer value.

**Comment by chuckharmston** [comment]
+1 to everything here. Gathering the actual data accurately is a logistical nightmare, and I think it's a better experience to give no information than incomplete or incorrect information, a lesson we've [learned before](https://bugzilla.mozilla.org/show_bug.cgi?id=1049770).

**Comment by dbialer** [comment]
I think you are right for the scenario that you describe: displaying on the app details page.  I do think website language has a lot of value, but agree that it doesn't have a lot of value in this particular use case. 

If we classify websites by country, the country parameter takes care of many of the language issues, except in multilingual countries (such as India).  This is not always dependable though, and as we (today) force a location on the user, it is not changeable.  This may be a problem for some cases.

I am not sure if you are suggesting that we don't have a way to sort-out language at all for the end user in some way - which could be by country.   Think of using Google Search without the implicit language or location filter. You would get sites in Chinese, Greek, Arabic, etc.   It is very easy in a browser to click on each one to see if it is in your language, and then close the tab if its not, but that seems annoying and not really catering to context of the user (which is the language that they understand).    I think we would be asking the user to do a lot of work to figure out relevance, without visual cues, implicit or explicit filtering to let them know if the site meets their need.   So in general, I think language of a website is important, but as a user I don't really care about the other languages a site is available in other than mine, and I don't really care how its accomplished.

If sites are classified by country, a lot of the problem is gone since presumably most US-relevant sites would be in English, most Germany-relevant sites would be in German, etc.  The only issues are countries that have large multi-lingual populations.

We could also possibly ask submitters to only submit for the language that they know about and can vouch for, and not ask them to list all the languages the site can be used in.   If they see it in English, they should say its in English and the description of the site should be in English.   

The site in Spanish can be a separate entry than the one in English with a separate description, even if they share the same URL.

**Comment by ezoehunt** [comment]
Yes we would filter out sites using the region exclusion. That's what I'm suggesting we do. It's a good and reliable indicator of whether a site would be relevant to me.

I want to get rid of the idea that we would show users the website languages. Because this information will be incomplete and/or inaccurate since we can't guarantee that we will get all the submissions that cover every language the site is available in.

If a website is submitted multiple time in different languages, that site should only appear once in Marketplace. I don't want to search for a site and see 25 listings that I have to read to find the one in the language I want. I simply want to click the URL and let the browser do the work for me - it knows the language I prefer, and the site knows to deliver that language if it has it. That's the beauty of the internet!

The way I picture it working is this:
- Site is submitted in English and Spanish. Each submitters submits the title/description in the language they see in the meta tags, so one set in English, one set in Spanish.
- A user who is viewing MP with browser set to "English" will see the English text. 
- User viewing MP with browser set to "Spanish" will see the Spanish text.
- Talk to Wil/Durst in more detail. The above is merely how I imagined it would work.

We're not going to be able to show a user only the listings that are relevant to him using "language" as a parameter. This is my understanding:
- Marketplace doesn't map languages to countries. So it doesn't know that X and Y are languages in India. 
- Creating such a mapping is work that we've not planned for. That work is very challenging and involves bias - how do you decide what languages should be attached to which countries? e.g., in a mixed ethnic country like the US or a multi-lingual country like India, which languages would you attach? 
- Also what do you do when a user has their browser set to a language that is not mapped to the region he's current in? E.g., I'm in the US and have Urdu set in my browser? Do you show 0 results for that user? That level of filtering, of "helping the user" -- is not useful. It involves cultural bias, and isn't on brand for Mozilla.
- For these reasons we decided to not move forward with [search by language](https://bugzilla.mozilla.org/show_bug.cgi?id=1111643). It's complicated and requires much more thought and work to make it work well for users.

**Comment by dbialer** [comment]
OK, I get what your saying.  Sounds good.  It sounds like can get localized Titles and descriptions through the collection process, but rely on using the browser language setting to show localized titles and descriptions should they exist.  But that is completely independent of country.

This sounds consistent with our current implementation for apps too.

**Comment by wfwalker** [comment]
Just to pile onto the original description -- websites don't sit still, either. If you publish a list of languages they support, you're committing to revisiting that site on an ongoing basis to recheck whether all those languages are still supported.

**Comment by ezoehunt** [comment]
Closing this discussion for now. 

In the future we may be able to figure out ways to deal with the issues we've discussed here. For now language isn't critical - we can rely on the region exclusions to help people find the right content. 

Because we _might_ be able to figure out to use language in the future, @dbialer and I would like to collect it from submitters. That way we'll have it if we decide to use it in the future. But we won't show this on the front-end to consumers.
